lib/plugins/rotate_asg_instances/asg.rb

```diff
@@ -48,6 +63,22 @@ def outdated_instances
 
     private
 
+    def ssh_executed_successfully?
+      Retriable.retriable(base_interval: 5, tries: 3) do
+        @command_response = run_ssh_command(first_ec2_instance, '/bin/true')
+        @command_response.exitstatus.zero?
```

Comment:
Retriable gem afaik on retries when it's block raises an error. Since this is block will only return false when command return non zero exitstatus it will not be retried.